### PR TITLE
[Improvement][E2E] support e2e compose v2 fix code style

### PR DESCRIPTION
**Pull request by jackyyyyyssss**
## Purpose of the pull request
The old version of testcontainers does not support Docker Compose v2, but currently, Docker Desktops in Windows/mac use v2, which causes local e2e runs to fail. Please refer to the specific issue for details 
https://github.com/testcontainers/testcontainers-java/issues/1481
  Update the testcontainers version for e2e testing and use the new API
https://github.com/testcontainers/testcontainers-java/pull/5608

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by jackyyyyyssss** [comment]
image
image
These errors don't seem to be within my scope of modification this time

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 1]
```diff
@@ -126,8 +127,8 @@ private void runInLocal() {
     private void runInDockerContainer(ExtensionContext context) {
         compose = createDockerCompose(context);
         compose.start();
-
-        address = HostAndPort.fromParts("host.testcontainers.internal", compose.getServicePort(serviceName, DOCKER_PORT));
+        address =
+                HostAndPort.fromParts("host.testcontainers.internal", compose.getServicePort(serviceName, DOCKER_PORT));
```
Comment (SbloodyS) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: Please do not add meaningless modifications.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 2]
```diff
@@ -194,22 +195,24 @@ private void setDriver(Object object, Field field) {
...
+                .collect(Collectors.toList());
+        ComposeContainer compose = new ComposeContainer(files)
+                .withPull(true)
+                .withTailChildContainers(true)
+                .withLocalCompose(true)
+                .withExposedService(
+                        serviceName,
+                        DOCKER_PORT, Wait.forListeningPort().withStartupTimeout(Duration.ofSeconds(300)))
+                .withLogConsumer(serviceName, outputFrame -> LOGGER.info(outputFrame.getUtf8String()))
+                .waitingFor(serviceName, Wait.forHealthcheck().withStartupTimeout(Duration.ofSeconds(300)));
```
Comment (SbloodyS) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: Same here.

File: dolphinscheduler-e2e/pom.xml [hunk 3]
```diff
@@ -43,8 +42,34 @@
         <slf4j-api.version>1.7.36</slf4j-api.version>
         <log4j-slf4j-impl.version>2.17.2</log4j-slf4j-impl.version>
         <guava.version>31.0.1-jre</guava.version>
+
+
```
Comment (SbloodyS) [inline dolphinscheduler-e2e/pom.xml]: Same here.

File: dolphinscheduler-e2e/pom.xml [hunk 4]
```diff
@@ -15,15 +15,14 @@
...
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>org.apache.dolphinscheduler</groupId>
     <artifactId>dolphinscheduler-e2e</artifactId>
-    <packaging>pom</packaging>
     <version>1.0-SNAPSHOT</version>
+    <packaging>pom</packaging>
```
Comment (SbloodyS) [inline dolphinscheduler-e2e/pom.xml]: Same here.

File: dolphinscheduler-e2e/pom.xml [hunk 5]
```diff
@@ -15,15 +15,14 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   -->
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (SbloodyS) [inline dolphinscheduler-e2e/pom.xml]: Same here.

File: dolphinscheduler-e2e/pom.xml [hunk 6]
```diff
@@ -43,8 +42,34 @@
...
+            </dependency>
+            <dependency>
+                <groupId>org.testcontainers</groupId>
+                <artifactId>testcontainers-bom</artifactId>
+                <version>1.19.3</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
```
Comment (SbloodyS) [inline dolphinscheduler-e2e/pom.xml]: Same here.

**Review by SbloodyS — CHANGES_REQUESTED** [review]

**Comment by jackyyyyyssss** [comment]
@SbloodyS  Thank you for your review. The reason for my formatting change is that the e2e module did not introduce mvn spotless : apply

So I introduced the same rules as the main module locally to run the formatting code and fix the previous format irregularities. I have restored it to see if it is necessary to introduce mvn spotless in the e2e module to maintain consistency with the main module？

**Comment by ruanwenjun** [comment]
We also need to update `testcontainer` version in `dolphinscheduler-bom`

**Comment by jackyyyyyssss** [comment]
> We also need to update `testcontainer` version in `dolphinscheduler-bom`

done @ruanwenjun  PTAL

**Comment by jackyyyyyssss** [comment]
@SbloodyS  PTAL

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java [hunk 7]
```diff
@@ -203,14 +203,18 @@ private DockerComposeContainer<?> createDockerCompose(ExtensionContext context)
...
+	   
 
         return compose;
     }
-}
+}
```
Comment (SbloodyS) [inline dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: We should leave a blank line at the end of each file.

**Review by SbloodyS — DISMISSED** [review]
Generally LGTM. Just some NIT.

File: dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java (hunk 7 above)
Comment (jackyyyyyssss) [reply dolphinscheduler-e2e/dolphinscheduler-e2e-core/src/main/java/org/apache/dolphinscheduler/e2e/core/DolphinSchedulerExtension.java]: done

**Review by SbloodyS — APPROVED** [review]

**Pull request merged**